src/Pos/Client/Txp/Util.hs

```diff
@@ -174,27 +220,35 @@ makeRedemptionTx rsk txInputs = makeAbstractTx mkWit (map ((), ) txInputs)
             }
 
 type FlatUtxo = [(TxIn, TxOutAux)]
-type InputPicker = StateT (Coin, FlatUtxo) (Either Text)
+type InputPicker = StateT (Coin, FlatUtxo) (Either TxError)
 
--- | Given Utxo, desired source addresses and desired outputs, prepare lists
--- of correct inputs and outputs to form a transaction
-prepareInpsOuts ::
-       MonadError Text m
+-- | Given filtered Utxo, desired outputs and fee size,
+-- prepare correct inputs and outputs for transaction
+-- (and tell how much to send to remaining address)
+prepareTxRaw
+    :: MonadError TxError m
     => Utxo
-    -> NonEmpty Address
     -> TxOutputs
-    -> m (TxOwnedInputs Address, TxOutputs)
-prepareInpsOuts utxo addrs outputs = do
+    -> TxFee
+    -> m TxRaw
+prepareTxRaw utxo outputs (TxFee fee) = do
     when (totalMoney == mkCoin 0) $
-        throwError "Attempted to send 0 money"
+        throwTxError "Attempted to send 0 money"
     futxo <- either throwError pure $
-        evalStateT (pickInputs []) (totalMoney, sortedUnspent)
+        evalStateT (pickInputs []) (totalMoneyWithFee, sortedUnspent)
     case nonEmpty futxo of
-        Nothing       -> throwError "Failed to prepare inputs!"
-        Just inputsNE -> pure (map formTxInputs inputsNE, outputs)
+        Nothing       -> throwTxError "Failed to prepare inputs!"
+        Just inputsNE -> do
+            let totalTxAmount = unsafeSumTxOuts $ map snd inputsNE
+                trInputs = map formTxInputs inputsNE
+                trOutputs = outputs
+                trRemaining = totalTxAmount `unsafeSubCoin` totalMoneyWithFee
```

Comment:
@Martoon-00 Fee is not going anywhere, it just gets lost (if it will go to a change address, it wouldn't be _spent_, and it wouldn't effectively be a _fee_